src/store/index.js

```diff
@@ -38,11 +48,36 @@ export default new Vuex.Store({
 		isBackdropActive(state) {
 			return state.backdrop;
 		},
+		isMessageActive(state) {
+			return state.message;
+		},
+		textMessage(state) {
+			return state.message_text;
+		},
 	},
 	actions: {
 		setLoading(context, status) {
-			context.commit('SET_BACKDROP', status);
-			context.commit('SET_LOADING', status);
+			let active = undefined;
+			if (typeof status === 'boolean') {
+				active = status;
+			} else {
+				active = status.active;
+			}
+			context.commit('SET_BACKDROP', active);
+			context.commit('SET_LOADING', active);
+
+			if (status.info && status.infoMessage) {
+				context.commit('SET_INFO', status.info);
+				context.commit('SET_INFO_MESSAGE', status.infoMessage);
+			}
+
+			if (status === false) {
```

Comment:
i would put it to the `else` brach above (in case status is boolean, it would go to the else branch)- and remove the bakcdrop and loading  - its happening twice if you pass `false`